### ISSUE TITLE
"Failure has occurred while loading a type" only on Build

### ISSUE THREAD

**Issue by MicBoucinha** [proyecto26/RestClient#34]
Hi,

First of all, thanks a lot for the library, it has been really enjoyable to use it in my project so far.

I am having an issue and was able to reproduce it in a simple project ([https://drive.google.com/open?id=1pUHzQEhPWuEpV1nAer7jUEYJKfdGuosq](https://drive.google.com/open?id=1pUHzQEhPWuEpV1nAer7jUEYJKfdGuosq)). The build version is also included. When in the Editor, everything works as expected. However, when I build the project, I receive a "Failure has occurred while loading a type." message and sometimes a message stating it cannot access the web address. I know that the request reaches the server (I confirmed it in the server's logs) and it seems that something is going wrong in the data processing of the response somehow. For other libraries that I use in my project, I have unity set to use the Scripting Runtime version .NET 4.x equivalent and the API Compatibility Level as .NET Standard 2.0.

I also tried to disable the firewall and the same thing happen.

Unfortunately I wasn't able to get a more detailed message or could pinpoint the error. Has anyone had a similar issue? If so, how did you solve it? Am I doing something wrong?

Thank you.

**Comment by jdnichollsc** [comment]
Hello, thanks for report this issue!
What platform are you building for?

**Comment by jdnichollsc** [comment]
Also, can you share the logs please?

**Comment by MicBoucinha** [comment]
Hi,

I'm building for Windows (I'm currently on win10 pro version 1803).

I'm attaching the log for that build in the zip file I sent before.
[output_log.txt](https://github.com/proyecto26/RestClient/files/2431470/output_log.txt)

**Comment by MicBoucinha** [comment]
It seems that I forgot to mention my Unity version (2018.1.4f1). Sorry about that.

**Comment by jdnichollsc** [comment]
@MicBoucinha Is that version Beta, right?

**Comment by jdnichollsc** [comment]
@MicBoucinha It looks like an Unity issue https://issuetracker.unity3d.com/issues/dot-net-4-dot-6-fails-to-load-types-if-the-source-code-file-size-is-larger-than-1-dot-95mb

**Comment by MicBoucinha** [comment]
Hi @jdnichollsc, as far as I know it is not a beta version. However, I will try to update it and confirm if the issue you mention is fixed. 

Thanks.

**Comment by jdnichollsc** [comment]
@MicBoucinha Can you attach a demo please? I can't reproduce this issue

**Comment by MicBoucinha** [comment]
@jdnichollsc The first link of google drive I gave in the first report is for a simple project where the problem occurs (for version 2018.1.4f1). I used the same project in the other versions to generate a Build of the project and give you the logs. The link is still valid at: [https://drive.google.com/open?id=1pUHzQEhPWuEpV1nAer7jUEYJKfdGuosq](https://drive.google.com/open?id=1pUHzQEhPWuEpV1nAer7jUEYJKfdGuosq).

In the zip file is also the Build version (Development mode) of the same project. When pressing the 'S' key, a GET request is sent to a public API.

If you need something else, please let me know. Thank you for the help.

**Comment by jdnichollsc** [comment]
Hi @MicBoucinha,
Check this comment https://github.com/proyecto26/RestClient/issues/35#issuecomment-428433327
Can you try removing **RSG.Promise_net35.dll** and later rename **RSG.Promise_standard.dll** to **RSG.Promise.dll**?
Please, let me know

**Comment by MicBoucinha** [comment]
Hi @jdnichollsc,

I tried your suggestion and unfortunately the same problem occurs. What is even stranger is that now I have different error messages for the 2018.1.6f1 and 2018.3b3 versions. It should be of note, that this only occurs when building. Inside the Editor, everything works as expected.

Also, and considering that the problem might lie in the [Nuget for Unity package ](https://github.com/GlitchEnzo/NuGetForUnity) that I'm using, which could be getting a wrong dll version, and for the RSG.Promise it was getting the .net standard version as expected, I decided to remove the library from there and import the .unitypackage that you have in the releases (and changed the name afterwards). The same problem occurs.

Here are the logs for these new 2 tests. Just like the previous logs I sent, this were made in a different machine which is basically "clean" (we only use it for data collection) to try to see if it could it be a configuration problem in my development machine.
[output_log Unity 2018.1.6f1.txt](https://github.com/proyecto26/RestClient/files/2464202/output_log.Unity.2018.1.6f1.txt)
[output_log Unity 2018.3b3.txt](https://github.com/proyecto26/RestClient/files/2464203/output_log.Unity.2018.3b3.txt)

Were you able to reproduce the problem in your end with the example I sent?

Thank you.

**Comment by jdnichollsc** [comment]
@MicBoucinha sorry man, I can't reproduce this issue in this moment because I'm using a Mac.
On the other hand, you're using beta versions of Unity, so you need to report this issue to the Unity team 👍 
PD: If you want, try the **[RSG.Promise](https://github.com/Real-Serious-Games/C-Sharp-Promise)** library directly and let me know if you can see the same error :)

**Comment by MicBoucinha** [comment]
Hi @jdnichollsc, Thanks. I will try to use RSG.Promise directly and let you know the results.

Regarding the beta versions, I only used it to verify if that issue you mentioned would be fixed at that point, since I'm not yet using the latest final version. The weird part of all of this to me is that it works perfectly fine inside the Editor.

All the best :) and thanks for your help.

**Comment by MicBoucinha** [comment]
Hi @jdnichollsc, I managed to "fix" the situation (and I'm embarrassed that took me this long). 

When trying to get more detail on the error that I was receiving I got to a point that mentioned a Newtonsoft.Json method that was throwing the exception. I found it strange but tried different approaches (even using the Unity version that someone created). No success with that version. However, that forced me to uninstall the official Newtonsoft.Json nuget package from NuGet for Unity. 

Afterwards, and since the other Json library wasn't working, I reinstalled the official version through nuget. And everything started working afterwards. It was really strange. Within the Editor everything was always working. When building, stuff simply didn't work. 

Maybe it was a cached version that allowed to work inside the Editor, or something else. Either way, after reinstalling Newtonsoft.Json **after** installing your library, everything started working.

Thank you so much for your patience on trying to help me figure out the problem.
